internal/fs/buffer/writebuffer.go

```diff
@@ -0,0 +1,26 @@
+package buffer
+
+import "io"
+
+// WriteBuffer is an interface that buffers the data to be written to GCS during
+// the write flow.
+// WriteBuffer is used only in create new file flow with sequential writes and
+// at any point in time, only 2x of the configured buffer size is stored in the
+// write buffer.
+type WriteBuffer interface {
```

Comment:
@ashmeenkaur as discussed offline, let's keep only those changes in the PR and in the shared branch which have matured enough to be shared and reviewed. For now, I think you only need a Write function in the WriteBuffer interbace, let's remove the others.